### ISSUE TITLE
Use MUnit for testing

### ISSUE THREAD

**Issue by janstenpickle** [janstenpickle/trace4cats#347]

**Comment by catostrophe** [comment]
Btw, I managed to make all tests work on Scalatest. Do we really need to make such a huge migration?

**Comment by janstenpickle** [comment]
I don't think we should make the CE3 release depend on a migration, and I think I might do this gradually, using both for a little while then a full migration later on.

**Comment by catostrophe** [comment]
I mean, if we start the migration to MUnit now, we'll have to do it in both branches - `master` and `CE3`.

**Comment by janstenpickle** [comment]
I suppose it depends whether we want to maintain two versions for the differnt cats-effect versions. Otherwise I only plan on updating the tests that haven't been modified in the `CE3` branch.

I'd prefer not maintaining two BTW, I think it's probably better to cut a final CE2 release, the use CE3 from there.

**Comment by catostrophe** [comment]
Let's plan the version number, e.g. `0.10.0` for the last CE2 release and make a branch for it. If we need to do a hotfix we can release it for CE2 version as `0.10.*`. And the first CE3 release may be `0.11.0`.
